src/modules/job-manager/urgency.c

```diff
@@ -40,6 +37,26 @@
 
 #define MAXOF(a,b)   ((a)>(b)?(a):(b))
 
+static int reprioritize (struct job_manager *ctx, struct job *job)
```

Comment:
we could move it to `alloc.c`, I simply kept it here b/c a change via an urgency change is the only use case at the moment.

Also, note that (currently) I only support changing the priority of one job

```
    if (!(f = flux_rpc_pack (ctx->h,
                             "sched.prioritize",
                             FLUX_NODEID_ANY,
                             FLUX_RPC_NORESPONSE,
                             "{s:[[I,I]]}",
                             "jobs",
                             job->id,
                             job->priority))) {
```

b/c that's the only use case right now.

I wasn't sure how "batch priority update" would occur down the road, so I sort of intentionally didn't want to put this function in `alloc.c` or a new `priority.c` as the priority plugin architecture code is still TBD.

Reply:
Maybe @grondo could comment here if one way or the other would be more convenient for his prototype plugin framework?

Reply:
I could go either way for this PR. Since we're still not sure how the full job queue re-prioritization will work, it isn't clear where this function belongs (though certainly, long-term `urgency.c` is the wrong place). It will probably be easier to see this once we figure out the method by which job reprioritization is driven. (At which point probably the function would be replaced with something different)

More generally, we should maybe thing about centralizing where priority updates are made and/or detected. Right now a lot is happening in `urgency_handle_request()` that feels like it maybe should be event driven in `event.c` or at least abstracted in some functions that could be called from alternate locations (e.g. a priority update loop).

Even that refactoring could be saved for the PR in which we implement the plugin architecture though.

Reply:
> I could go either way for this PR. Since we're still not sure how the full job queue re-prioritization will work, it isn't clear where this function belongs (though certainly, long-term urgency.c is the wrong place).

Agreed, `urgency.c` is not the right place long term.  Just left it there b/c that's the one use case so far.  So I think I'll leave it there for now.